include/envoy/network/io_handle.h

```diff
@@ -1,33 +1,136 @@
 #pragma once
 
+#include <bitset>
 #include <memory>
 
 #include "envoy/common/pure.h"
 
 namespace Envoy {
+
+namespace Buffer {
+struct RawSlice;
+}
+
 namespace Network {
 
+namespace Address {
+class Instance;
+} // namespace Address
+
+class IoError {
+public:
+  enum class IoErrorCode {
```

Comment:
if this merges after #5772, can we also update that API to reference IOError here rather than do what it's doing?

Also IMO IOError should not be underneath the Network namespace, as this also makes sense for non-networked I/O.